src/reporter.ts

```diff
@@ -383,6 +337,53 @@ export default class Reporter {
 
     return run;
   }
+
+  collectAttachments(result: RunResult) {
+    const attachments: Attachment[] = [];
+    if (result.video) {
+      attachments.push({
+        name: 'video',
+        path: VIDEO_FILENAME,
+        contentType: 'video/mp4',
+      });
+    }
+    result.screenshots?.forEach((s) => {
+      attachments.push({
+        name: 'screenshot',
+        path: path.basename(s.path),
+        contentType: 'image/png',
+      });
+    });
+    return attachments;
+  }
+
+  collectAssets(result: RunResult, report: TestRun): Asset[] {
```

Comment:
These methods need docs. What's the difference between an attachment and an asset? One is defined by test composer, the other by the json reporter. But I really shouldn't have to visit these libraries just to figure out what it means in the context used here.